src/shogun/regression/GLM.cpp

```diff
@@ -0,0 +1,82 @@
+#include <shogun/lib/config.h>
+
+#include <shogun/labels/RegressionLabels.h>
+#include <shogun/lib/SGVector.h>
+#include <shogun/mathematics/Math.h>
+#include <shogun/mathematics/linalg/LinalgNamespace.h>
+#include <shogun/regression/GLM.h>
+#include <utility>
+using namespace shogun;
+
+GLM::GLM() : LinearMachine()
+{
+	init();
+}
+SGVector<float64_t> log_likelihood(
+    const std::shared_ptr<DenseFeatures<float64_t>> features,
+    std::shared_ptr<Labels> label)
+{
+	int32_t vector_count = features->get_num_vectors();
+	int32_t feature_count = features->get_num_features();
+	ASSERT(vector_count > 0 && label->get_num_labels() == vector_count)
+	// Array of Lambdas
+	SGVector<float64_t> lambda(vector_count);
+	for (int32_t i = 0; i < vector_count; i++)
+	{
+		SGVector<float64_t> feature_vector = features->get_feature_vector(i);
+		// Assume beta is the same as the feature vector
+		SGVector<float64_t> beta = feature_vector.clone();
+		// Assume beta0 is the same as the first element in the feature vector
+		float64_t b0 = feature_vector.get_element(0);
+		SGVector<float64_t> result(feature_count);
+		SGVector<float64_t>::vector_multiply(
+		    result, feature_vector, beta, feature_vector.vlen);
+		float64_t res = SGVector<float64_t>::sum(result);
+		lambda.set_element(log(1 + std::exp(b0 + res)), i);
+	}
+	SGVector<float64_t> likelihood(vector_count);
+
+	for (int32_t i = 0; i < vector_count; i++)
```

Comment:
this whole thing does the following:
likelihood = y*log(lambda) - lambda
this is definitely vectorizable... please use linalg framework to do this. instead of doing this vanilla code